### PR TITLE
8234 - Payment Information

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:* /bcgov/8234

*Description of changes:*

- Adding payment info component
- test cases

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/ViewRoutingSlip.vue [hunk 1]
```diff
@@ -23,25 +23,29 @@
...
-    RoutingSlipInfo
+    RoutingSlipInfo,
+    PaymentInformation
   },
   setup (props) {
     useViewRoutingSlip(props)
   }
 })
-export default class CreateRoutingSlipView extends Vue {
+export default class ViewRoutingSlipView extends Vue {
```
Comment (sumesh-aot) [inline src/views/ViewRoutingSlip.vue:48]: any better names? 2 `View`

File: src/components/ViewRoutingSlip/PaymentInformation.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<template>
+  <div >
+    <h3 data-test="title">02.Payment Information</h3>
+    <div class="d-flex justify-space-between">
+      <p class="ma-0 align-self-center">
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at
```
[the remark this answers is not in the document]
Comment (karthik-aot) [reply src/components/ViewRoutingSlip/PaymentInformation.vue:6]: Yeah sure. There are many places in Create Routing slip and other views that need wordings from Roy

File: src/views/ViewRoutingSlip.vue (hunk 1 above)
Comment (karthik-aot) [reply src/views/ViewRoutingSlip.vue]: good one. Renamed :)

File: src/components/RoutingSlip/CreateRoutingSlipCashPayment.vue [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <template>
-    <v-form ref="createRoutingSlipCashPaymentForm">
+    <v-form ref="createRoutingSlipCashPaymentForm" :disabled="isViewMode">
```
Comment (shabeeb-aot) [inline src/components/RoutingSlip/CreateRoutingSlipCashPayment.vue:2]: is it better to use v-can like sbc auth?

File: src/composables/ViewRoutingSlip/usePaymentInformation.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  // As per current business rule, a routingslip has one-to-one relation with payment method (Cash/Cheque)
+  // Therefore, we can determine the payment method of the current routingslip from the first payment record
+  const isPaymentCheque = computed(() => {
+    const payments = (routingSlip.value as RoutingSlip)?.payments
+    // to prevent lazy load
+    return payments && payments[0].paymentMethod === PaymentMethods.CHEQUE
+  })
+
+  function addAdditionalFunds (): void {
+  }
```
Comment (shabeeb-aot) [inline src/composables/ViewRoutingSlip/usePaymentInformation.ts:28]: unused function?

Comment (karthik-aot) [reply src/composables/ViewRoutingSlip/usePaymentInformation.ts:28]: For add additionalfunds. for future. But I can remove :)

File: src/components/RoutingSlip/CreateRoutingSlipCashPayment.vue (hunk 3 above)
Comment (karthik-aot) [reply src/components/RoutingSlip/CreateRoutingSlipCashPayment.vue:2]: since there is only page level access, we can ignore for now?

**Review by shabeeb-aot — APPROVED** [review]

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**